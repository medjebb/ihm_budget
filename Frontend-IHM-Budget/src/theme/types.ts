import type {
  Shadows,
  ColorSystemOptions,
  SupportedColorScheme,
  ThemeOptions as MuiThemeOptions,
} from '@mui/material/styles';

import type { CustomShadows } from './core/custom-shadows';

export type ThemeColorScheme = SupportedColorScheme;
export type ThemeCssVariables = {
  cssVarPrefix?: string;
  colorSchemeSelector?: string;
  disableCssColorScheme?: boolean;
  shouldSkipGeneratingVar?: (keys: string[]) => boolean;
};

type ColorSchemeOptionsExtended = ColorSystemOptions & {
  shadows?: Shadows;
  customShadows?: CustomShadows;
};

export type ThemeOptions = MuiThemeOptions & {
  defaultColorScheme?: ThemeColorScheme;
  colorSchemes?: Partial<Record<ThemeColorScheme, ColorSchemeOptionsExtended>>;
  cssVariables?: ThemeCssVariables;
};
