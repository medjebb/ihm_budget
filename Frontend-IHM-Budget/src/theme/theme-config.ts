import type { CommonColors } from '@mui/material/styles';

import type { ThemeCssVariables } from './types';
import type { PaletteColorNoChannels } from './core/palette';

type ThemeConfig = {
  classesPrefix: string;
};

export const themeConfig: ThemeConfig = {
  classesPrefix: 'min imal',
  // fontFamily: {
  //   primary: 'DM Sans Variable',
  //   secondary: 'Barlow',
  // },
  // palette: {
  // warning: {
  //   lighter: '#FFF5CC',
  //   light: '#FFD666',
  //   main: '#FFAB00',
  //   dark: '#B76E00',
  //   darker: '#7A4100',
  //   contrastText: '#1C252E',
  //   mainChannel: '255, 171, 0',
  //   lightChannel: '255, 214, 102',
  //   darkChannel: '183, 110, 0',
  //   contrastTextChannel: '28, 37, 46',
  // },
  // error: {
  //   lighter: '#FFE9D5',
  //   light: '#FFAC82',
  //   main: '#FF5630',
  //   dark: '#B71D18',
  //   darker: '#7A0916',
  //   contrastText: '#FFFFFF',
  //   mainChannel: '255, 86, 48',
  //   lightChannel: '255, 172, 130',
  //   darkChannel: '183, 29, 24',
  //   contrastTextChannel: '255, 255, 255',
  // },
  // grey: {
  //   '50': '#FCFDFD',
  //   '100': '#F9FAFB',
  //   '200': '#F4F6F8',
  //   '300': '#DFE3E8',
  //   '400': '#C4CDD5',
  //   '500': '#919EAB',
  //   '600': '#637381',
  //   '700': '#454F5B',
  //   '800': '#1C252E',
  //   '900': '#141A21',
  // },
  // common: { black: '#000000', white: '#FFFFFF' },
  // },
  // cssVariables: {
  //   cssVarPrefix: '',
  //   colorSchemeSelector: 'data-color-scheme',
  // },
};
