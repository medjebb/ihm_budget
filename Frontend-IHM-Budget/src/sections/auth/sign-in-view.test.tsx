import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'src/theme/theme-provider';
import { SignInView } from './sign-in-view';

jest.mock('src/routes/hooks', () => {
  const push = jest.fn();
  return {
    useRouter: () => ({ push }),
  };
});

const renderComponent = () =>
  render(
    <ThemeProvider>
      <MemoryRouter>
        <SignInView />
      </MemoryRouter>
    </ThemeProvider>
  );

describe('SignInView', () => {
  it('renders the sign-in form', () => {
    const { asFragment } = renderComponent();
    expect(screen.getByRole('heading', { name: /sign in/i })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('toggles password visibility', () => {
    renderComponent();
    const toggleButton = screen.getByTestId('toggle-password-visibility');
    const passwordInput = screen.getByLabelText(/password/i);

    expect(passwordInput).toHaveAttribute('type', 'password');

    fireEvent.click(toggleButton);
    expect(passwordInput).toHaveAttribute('type', 'text');

    fireEvent.click(toggleButton);
    expect(passwordInput).toHaveAttribute('type', 'password');
  });

  it('navigates to home on sign-in', () => {
    // eslint-disable-next-line global-require
    const { push } = require('src/routes/hooks').useRouter();
    renderComponent();
    const signInButton = screen.getByRole('button', { name: /sign in/i });

    fireEvent.click(signInButton);
    expect(push).toHaveBeenCalledWith('/');
  });
});
