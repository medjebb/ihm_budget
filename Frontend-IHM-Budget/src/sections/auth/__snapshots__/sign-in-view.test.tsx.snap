// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInView renders the sign-in form 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1t1x6d"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1xvf6pp-MuiTypography-root"
    >
      Sign in
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body2 css-drhbix-MuiTypography-root"
    >
      Donâ€™t have an account?
      <a
        class="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineHover css-3x996s-MuiTypography-root-MuiLink-root"
      >
        Get started
      </a>
    </p>
  </div>
  <div
    class="MuiBox-root css-1f3cjlq"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-32yp37-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1mib03-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r0:"
        id=":r0:-label"
      >
        Email address
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1faydbs-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-evvb5l-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          name="email"
          type="text"
          value="hello@gmail.com"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1nldwwb-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Email address
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <a
      class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-1roa2zs-MuiTypography-root-MuiLink-root"
    >
      Forgot password?
    </a>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-32yp37-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1mib03-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r1:"
        id=":r1:-label"
      >
        Password
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1pgxdx6-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1s8eoz8-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r1:"
          name="password"
          type="password"
          value="@demo1234"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-csfdax-MuiInputAdornment-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1gd7tmh-MuiButtonBase-root-MuiIconButton-root"
            data-testid="toggle-password-visibility"
            tabindex="0"
            type="button"
          >
            <span />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1nldwwb-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Password
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorInherit MuiButton-disableElevation MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorInherit MuiButton-disableElevation MuiButton-fullWidth css-j4ao3a-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Sign in
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1f6a2fz-MuiDivider-root"
    role="separator"
  >
    <span
      class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <span
        class="MuiTypography-root MuiTypography-overline css-qq2ntd-MuiTypography-root"
      >
        OR
      </span>
    </span>
  </div>
  <div
    class="MuiBox-root css-1jtf595"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1y3h6m1-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <span />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
